specs/book_spec.rb

```diff
@@ -0,0 +1,18 @@
+require_relative '../book'
+require_relative '../rental'
+
+describe Book do
+  describe '#add_rental' do
+    it 'adds a rental to the book and person' do
+      person = double('Person')
+      allow(person).to receive(:add_rental)
+      book = Book.new('Title', 'Author')
+      date = '2023-01-01'
+
+      book.add_rental(person, date)
+
+      expect(book.rentals).not_to be_empty
+      expect(person).to have_received(:add_rental).with(an_instance_of(Rental))
+    end
+  end
```

Comment:
- [ ] @nilab-khwaja and @Izebiafe Kindly consider adding tests for all methods in the Book class, you are missing tests for `initialize` and `to_json`, kindly consider adding them:

![image](https://github.com/nilab-khwaja/OOP-school-library/assets/61408860/78df33f8-ce97-4681-9005-dff0e65fd636)